packages/test/test-service-load/src/gcDataStores.ts

```diff
@@ -164,55 +194,92 @@ export class DataObjectNonCollab extends BaseDataObject implements IGCDataStore
     /**
      * The map that stores the fluid handles to all child data objects.
      * Note: This should not be called before "run" is called which initializes it.
      */
-    private _childMap: SharedMap | undefined;
-    protected get childMap(): SharedMap {
-        assert(this._childMap !== undefined, "Child map cannot be retrieving before initialization");
-        return this._childMap;
+    private _dataObjectMap: SharedMap | undefined;
+    protected get dataObjectMap(): SharedMap {
+        assert(this._dataObjectMap !== undefined, "Child map cannot be retrieving before initialization");
+        return this._dataObjectMap;
+    }
+
+    private readonly blobMapKey = "blobMap";
+    /**
+     * The map that stores the fluid handles to all attachment blobs.
+     * Note: This should not be called before "run" is called which initializes it.
+     */
+    private _blobMap: SharedMap | undefined;
+    protected get blobMap(): SharedMap {
+        assert(this._blobMap !== undefined, "Blob map cannot be retrieving before initialization");
+        return this._blobMap;
     }
 
-    private readonly unreferencedChildrenDetails: IUnreferencedChildDetails[] = [];
-    private readonly referencedChildrenDetails: IChildDetails[] = [];
+    private readonly unreferencedChildDataObjects: IActivityObjectDetails[] = [];
+    private readonly referencedChildDataObjects: IActivityObjectDetails[] = [];
+
+    private readonly unreferencedAttachmentBlobs: IActivityObjectDetails[] = [];
+    private readonly referencedAttachmentBlobs: IActivityObjectDetails[] = [];
 
     protected async initializingFirstTime(): Promise<void> {
         await super.initializingFirstTime();
-        this.root.set<IFluidHandle>(this.childMapKey, SharedMap.create(this.runtime).handle);
+        this.root.set<IFluidHandle>(this.dataObjectMapKey, SharedMap.create(this.runtime).handle);
+        this.root.set<IFluidHandle>(this.blobMapKey, SharedMap.create(this.runtime).handle);
     }
 
     protected async hasInitialized(): Promise<void> {
         await super.hasInitialized();
-        const handle = this.root.get<IFluidHandle<SharedMap>>(this.childMapKey);
-        assert(handle !== undefined, "The child map handle should exist on initialization");
-        this._childMap = await handle.get();
+        const dataObjectMapHandle = this.root.get<IFluidHandle<SharedMap>>(this.dataObjectMapKey);
+        assert(dataObjectMapHandle !== undefined, "The child map handle should exist on initialization");
+        this._dataObjectMap = await dataObjectMapHandle.get();
 
-        // Initialize the referenced children list from the child map.
-        for (const childDetails of this._childMap) {
+        const blobMapHandle = this.root.get<IFluidHandle<SharedMap>>(this.blobMapKey);
+        assert(blobMapHandle !== undefined, "The blob map handle should exist on initialization");
+        this._blobMap = await blobMapHandle.get();
```

Comment:
When do we run?

Reply:
This is just getting the map where we store the handles of attachment blobs. We will run blobs when we initialize, create a new blob or revive an unreferenced blob.